### PR TITLE
Reports Updates

### PR DESCRIPTION
**Pull request by dparker1005**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Copy of #2394 built of the dev branch instead of 3.0.

- Adds "compare to previous period" for sales report
- Adds predefined timeframes for sales report
- Adds "Active Members Per Level" report
- Updates UI for logins report
Note: Logins report is not 100% MMPU compatible. Only issue is that the "Level" column only shows one level for each user. This may not be too big of an issue, but something to keep in mind.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by MaximilianoRicoTabo — APPROVED** [review]
Tested at local and looks and works great. I'm leaving few comments in the code despite i understand the PR actually gathers a bunch of different commits from different people in a wide span of time and then not expecting we spend time on it unless we want to do it. Perhaps we can  apply some of theme into v3.

File: adminpages/reports.php [hunk 1]
```diff
@@ -15,9 +15,22 @@
...
+	<ul class="subsubsub">
+		<li><a href="<?php echo esc_url( admin_url( 'admin.php?page=pmpro-reports' ) ); ?>"><?php esc_html_e('All', 'paid-memberships-pro' ); ?></a></li>
+		<?php foreach ( $pmpro_reports as $report_menu_item => $report_menu_title ) {
+			if ( function_exists( 'pmpro_report_' . $report_menu_item . '_page' ) ) { ?>
+				<li>&nbsp;|&nbsp;<a class="<?php if ( $report === $report_menu_item ) { ?>current<?php } ?>"href="<?php echo esc_url( admin_url( 'admin.php?page=pmpro-reports&report=' . $report_menu_item ) ); ?>"><?php echo $report_menu_title; ?></a></li>
+				<?php
+			}
+		} ?>
+	</ul>
+	<br class="clear" />
```
[the remark this answers is not in the document]
Comment (ideadude) [reply adminpages/reports.php:29]: Good point. In this case, I think consistency with how the same submenu is coded/styled on the advanced settings page makes it okay to continue using this method. Kim might catch stuff like this in the general style update pass.

File: adminpages/reports/sales.php [hunk 2]
```diff
@@ -145,175 +214,354 @@ function pmpro_report_sales_page()
 		$discount_code = '';
 	}
 
-	$currently_in_period = false;
+	if ( isset( $_REQUEST[ 'show_parts' ] ) ) {
+		$new_renewals = sanitize_text_field( $_REQUEST[ 'show_parts' ] );
+	} else {
+		$new_renewals = 'new_renewals';
+	}
```
[the remark this answers is not in the document]
Comment (ideadude) [reply adminpages/reports/sales.php:221]: I leave these to coder's discretion. I feel writing these out over a few lines is sometimes more readable than a single line ternary operator, but that's subjective. We don't generally enforce either way as a coding standard.

**Comment by ideadude** [comment]
It's also unclear what date range the "previous period" is. We could add a note about it at the bottom where we mention the average line. Or make sure to include it in the hover on individual dates/bars/points or a hover on the legend.

**Comment by dparker1005** [comment]
@ideadude I couldn't replicate the extra month showing on the monthly report (tried setting timezone to UTC-12 and UTC+14) or the wrap issue (moves the chart down correctly on my site). I did address the "previous period" confusion by changing the text to "previous year" though, since the previous period right now is always the previous year

**Comment by ideadude** [comment]
I believe this was the correct fix: https://github.com/strangerstudios/paid-memberships-pro/commit/312a88e82df9fa86b37ec9e97ed17f6ed5219fa9

It fixes my site. I believe there needs to be an order on Jan 1 for this to happen.

I already merged this PR, but double check that that change won't break other things.

**Pull request merged**